mod/custom_index/views/default/custom_index/content.php

```diff
@@ -6,79 +6,86 @@
  * page-rendering helper functions like elgg_view_page.
  */
 
-$list_params = [
-	'type' => 'object',
-	'limit' => 4,
-	'pagination' => false,
-	'no_results' => true,
-];
-
-?>
-
-<div class="custom-index elgg-main elgg-grid clearfix">
-	<div class="elgg-col elgg-col-1of2 custom-index-col1">
-		<div class="elgg-inner">
-<?php
-// left column
-
-// Top box for login or welcome message
-if (elgg_is_logged_in()) {
-	$content = elgg_format_element('h2', [], elgg_echo('welcome') . ' ' . elgg_get_logged_in_user_entity()->getDisplayName());
-	
-	echo elgg_view_module('featured', '', $content);
-} else {
-	echo elgg_view_module('featured', elgg_echo('login'), elgg_view_form('login'));
+$plugin = elgg_get_plugin_from_id('custom_index');
+$is_module_enabled = function(string $module) use ($plugin) {
+	return (bool) $plugin->{"module_{$module}_enabled"};
+};
+
+$get_list_params = function(array $extras = []) {
+	return array_merge([
+		'type' => 'object',
+		'limit' => 4,
+		'pagination' => false,
+		'no_results' => true,
+	], $extras);
+};
+
+$modules = [];
+
+if ($is_module_enabled('about')) {
+	$about = $plugin->about;
+	if (!empty($about)) {
+		$modules[] = elgg_view_module('featured', '', $about);
+	}
+}
+
+if (!elgg_is_logged_in() && $is_module_enabled('register')) {
+	$modules[] = elgg_view_module('featured', elgg_echo('register'), elgg_view_form('register'));
+}
+
+if (!elgg_is_logged_in() && $is_module_enabled('login')) {
+	$modules[] = elgg_view_module('featured', elgg_echo('login'), elgg_view_form('login'));
 }
 
 // a view for plugins to extend
-echo elgg_view('index/lefthandside');
+$lefthandside = elgg_view('index/lefthandside');
+if (!empty($lefthandside)) {
+	$modules[] = elgg_view_module('inline', '', $lefthandside);
```

Comment:
wouldn't this possibly cause modules in a module?

Reply:
changed